waspc/src/Wasp/Generator/Job/Process.hs

```diff
@@ -88,14 +90,14 @@ runNodeCommandAsJob fromDir command args jobType chan = do
   case errorOrNodeVersion of
     Left errorMsg -> exitWithError (ExitFailure 1) (T.pack errorMsg)
     Right nodeVersion ->
-      if nodeVersion < C.nodeVersion
-        then
-          exitWithError
-            (ExitFailure 1)
-            (T.pack $ "Your node version is too low. " ++ waspNodeRequirementMessage)
-        else do
+      if isVersionInBounds nodeVersion C.nodeVersionBounds
```

Comment:
@Martinsos 

This line is probably the main argument on why it probably makes sense to introduce both `SemanticVersion` and `VersionBounds` (perhaps I should rename it to `SemanticVersionBounds`?).

If we change our requirements for the node version, we don't need to worry about updating the condition here (e.g., `<` or `<=` or something else)

Reply:
Yeah, cool!
I commented on the naming below.